sound/soc/sof/amd/acp.h

```diff
@@ -122,6 +135,7 @@ struct acp_dev_data {
 	dma_addr_t dma_addr;
 	u8 *data_buf;
 	struct dma_descriptor dscr_info[ACP_MAX_DESC];
+	struct acp_dsp_stream *stream_buf[ACP_MAX_STREAM];
```

Comment:
yes, looks better, but it seems to me you could even make it an array of objects directly and avoid allocating them individually?

Reply:
in the Intel parts we allocate the streams dynamically because the numbers for playback and capture vary between platforms, but indeed here if it's constant and not subject to change you may want to skip the individual allocation.